File: app/src/main/java/com/doubean/ford/data/vo/GroupItem.java
package com.doubean.ford.data.vo;

import androidx.annotation.NonNull;
import androidx.room.PrimaryKey;

import com.google.gson.annotations.SerializedName;

import java.time.LocalDateTime;

public class GroupItem {
    @PrimaryKey
    @NonNull
    public String id;

    @NonNull
    public String name;

    @SerializedName("member_count")
    public Integer memberCount;

    @SerializedName("topic_count")
    public Integer postCount;

    @SerializedName("create_time")
    public LocalDateTime dateCreated;

    @SerializedName("sharing_url")
    public String url;

    @SerializedName("avatar")
    public String avatarUrl;

    @SerializedName("member_name")
    public String memberName;

    @SerializedName("desc_abstract")
    public String shortDescription;

    public GroupItem() {
    }

}

File: app/src/main/java/com/doubean/ford/data/vo/GroupItem.kt
package com.doubean.ford.data.vo

import androidx.room.PrimaryKey
import com.google.gson.annotations.SerializedName
import java.time.LocalDateTime

class GroupItem (
    @PrimaryKey
    val id: String,

    val name: String,

    @SerializedName("member_count")
    val memberCount: Int? = null,

    @SerializedName("topic_count")
    val postCount: Int? = null,

    @SerializedName("create_time")
    val dateCreated: LocalDateTime? = null,

    @SerializedName("sharing_url")
    val url: String,

    val uri: String,

    @SerializedName("avatar")
    val avatarUrl: String? = null,

    @SerializedName("member_name")
    val memberName: String? = null,

    @SerializedName("desc_abstract")
    val shortDescription: String? = null
)